### PR TITLE
[Offload] Provide a CMake cache file to easily build offloading

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
This patch adds a cache file that will automatically enable openpm,
offload, and all the fancy GPU libraries.

### CODE REVIEW

File: offload/cmake/caches/Offload.cmake [hunk 1]
```diff
@@ -0,0 +1,12 @@
+set(LLVM_ENABLE_PROJECTS "clang;clang-tools-extra;lld" CACHE STRING "")
```
Comment (shiltian) [inline offload/cmake/caches/Offload.cmake:1]: FWIW, `compiler-rt` is also required sometimes, such as HIP.

Comment (jhuber6) [reply offload/cmake/caches/Offload.cmake:1]: Why does HIP require the `clang` compiler-rt? Does it also enable `-frtlib=compiler-rt`? I can add it, but one unfortunate thing about having `compiler-rt` enabled in the main project is that it forces a CMake reconfigure *every* single time you run `ninja`.

File: openmp/docs/SupportAndFAQ.rst [hunk 2]
```diff
@@ -51,6 +51,22 @@ All patches go through the regular `LLVM review process
...
+  $> cd llvm-project  # The llvm-project checkout
+  $> mkdir build
+  $> cd build
+  $> cmake ../llvm -G Ninja                                                 \
+     -C ../offload/cmake/caches/Offload.cmake \ # The preset cache file
+     -DCMAKE_BUILD_TYPE=<Debug|Release>   \ # Select build type
+     -DCMAKE_INSTALL_PREFIX=<PATH>        \ # Where the libraries will live
+  $> ninja install
+
 To build an *effective* OpenMP offload capable compiler, only one extra CMake
```
Comment (shiltian) [inline openmp/docs/SupportAndFAQ.rst:70]: This needs to be refined in a way that, as a manual alternative, ...

**Review by shiltian — COMMENTED** [review]
The change looks okay

File: offload/cmake/caches/Offload.cmake (hunk 1 above)
Comment (shiltian) [reply offload/cmake/caches/Offload.cmake:1]: I'm not sure why `compiler-rt` is needed but it definitely doesn't work if it is not enabled.

> but one unfortunate thing about having compiler-rt enabled in the main project is that it forces a CMake reconfigure every single time you run ninja.

I always have it enabled but didn't see that issue.

Comment (jhuber6) [reply offload/cmake/caches/Offload.cmake:1]: Only when it's enabled as a runtime.

**Review by shiltian — APPROVED** [review]
LGTM. We can always improve it if necessary.

**Pull request merged**